src/coreclr/gc/gc.cpp

```diff
@@ -20582,7 +20819,126 @@ void gc_heap::garbage_collect (int n)
 #define mark_stack_empty_p() (mark_stack_base == mark_stack_tos)
 
 inline
-size_t& gc_heap::promoted_bytes(int thread)
+size_t gc_heap::get_promoted_bytes()
+{
+#ifdef USE_REGIONS
+    if (!survived_per_region)
+    {
+        dprintf (REGIONS_LOG, ("no space to store promoted bytes"));
+        return 0;
+    }
+
+    dprintf (3, ("h%d getting surv", heap_number));
+    size_t region_count = get_total_region_count();
+    size_t promoted = 0;
+    for (int i = 0; i < region_count; i++)
```

Comment:
how many regions do we expect for a typical app? Is there a way to optimize the loop?

Reply:
by default each region is 4mb, so 1k for a 4gb heap. we are just going linearly though a block of memory which is very fast. I can certainly see it's more optimal to only update the regions we actually use (when we have most regions not used it'll definitely be faster but that'll soon change as our heap range will be what we actually use instead of the total reserve) like I do in `sync_promoted_bytes`. I'll leave these optimizations for another PR - these changes have stayed local for long enough...  I'll add this to my TODO list.

Reply:
sounds good.